### PR TITLE
Merge First. BTC Inline

### PR DESCRIPTION
**Pull request by spencerlievens**

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -159,26 +159,25 @@ CDynamicAddress GetAccountAddress(std::string strAccount, bool bForceNew=false)
...
-        CScript scriptPubKey = GetScriptForDestination(account.vchPubKey.GetID());
-        for (std::map<uint256, CWalletTx>::iterator it = pwalletMain->mapWallet.begin();
-             it != pwalletMain->mapWallet.end() && account.vchPubKey.IsValid();
-             ++it)
-        {
-            const CWalletTx& wtx = (*it).second;
-            BOOST_FOREACH(const CTxOut& txout, wtx.vout)
-                if (txout.scriptPubKey == scriptPubKey)
-                    bKeyUsed = true;
+    if (!bForceNew) {
```
Comment (AmirAbrams) [inline src/wallet/rpcwallet.cpp:162]: I don't see this code in Bitcoin or Dash.  Is it custom code?

File: src/Makefile.new.am [hunk 2]
```diff
@@ -129,7 +129,6 @@ DYNAMIC_H = \
   keepass.h \
   keystore.h \
   limitedmap.h \
-  main.h \
```
Comment (AmirAbrams) [inline src/Makefile.new.am:132]: I think we can delete Makefile.new.am.

File: src/validation.h [hunk 3]
```diff
@@ -305,7 +305,7 @@ void PruneAndFlush();
 
 /** (try to) add transaction to memory pool **/
 bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransaction &tx, bool fLimitFree,
-                        bool* pfMissingInputs, bool fOverrideMempoolLimit=false, bool fRejectAbsurdFee=false, bool fDryRun=false);
+                        bool* pfMissingInputs, bool fOverrideMempoolLimit=false, const CAmount nAbsurdFee=0, bool fDryRun=false);
```
Comment (AmirAbrams) [inline src/validation.h:312]: I think Bitcoin removes the default value for nAbsurdFee

File: src/dynode.h [hunk 4]
```diff
@@ -203,8 +203,8 @@ class CDynode : public dynode_info_t
 
     bool UpdateFromNewBroadcast(CDynodeBroadcast& dnb, CConnman& connman);
 
-    static CollateralStatus CheckCollateral(const COutPoint& outpoint);
-    static CollateralStatus CheckCollateral(const COutPoint& outpoint, int& nHeightRet);
+    static CollateralStatus CheckCollateral(const COutPoint& outpoint, const CPubKey& pubkey);
```
Comment (AmirAbrams) [inline src/dynode.h:207]: What version or branch of Dash did this come from?

File: src/dynode.h [hunk 5]
```diff
@@ -149,7 +149,8 @@ class CDynode : public dynode_info_t
     enum CollateralStatus {
         COLLATERAL_OK,
         COLLATERAL_UTXO_NOT_FOUND,
-        COLLATERAL_INVALID_AMOUNT
+        COLLATERAL_INVALID_AMOUNT,
+        COLLATERAL_INVALID_PUBKEY
```
Comment (AmirAbrams) [inline src/dynode.h:153]: What is COLLATERAL_INVALID_PUBKEY used for.  I don't see it referenced.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (spencerlievens) [reply src/wallet/rpcwallet.cpp:162]: https://github.com/dashpay/dash/pull/1770/files

File: src/Makefile.new.am (hunk 2 above)
Comment (spencerlievens) [reply src/Makefile.new.am:132]: https://github.com/duality-solutions/Dynamic-2/pull/104

File: src/dynode.h (hunk 4 above)
Comment (spencerlievens) [reply src/dynode.h]: develop branch of Dash

File: src/validation.h (hunk 3 above)
Comment (spencerlievens) [reply src/validation.h:312]: https://github.com/dashpay/dash/pull/1770/files

File: src/dynode.h (hunk 5 above)
Comment (spencerlievens) [reply src/dynode.h]: dynode.cpp and governance-object.cpp

Comment (spencerlievens) [reply src/dynode.h:153]: https://github.com/dashpay/dash/pull/1783/files

File: src/dynode.h (hunk 4 above)
Comment (spencerlievens) [reply src/dynode.h:207]: https://github.com/dashpay/dash/pull/1783/files

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (AmirAbrams) [reply src/wallet/rpcwallet.cpp:162]: Alright, looks like a tested Dash PR.

File: src/validation.h (hunk 3 above)
Comment (AmirAbrams) [reply src/validation.h:312]: Bitcoin added it with the default = 0 and later removed the default:
https://github.com/bitcoin/bitcoin/blob/master/src/validation.h#L310

We can remove the default later.  I traced the code and it shouldn't cause any harm.

**Review by AmirAbrams — APPROVED** [review]

**Pull request merged**